modyn/supervisor/internal/triggers/trigger.py

```diff
@@ -1,20 +1,37 @@
+import pathlib
 from abc import ABC, abstractmethod
+from typing import Generator
 
 
 class Trigger(ABC):
     def __init__(self, trigger_config: dict) -> None:
         assert trigger_config is not None, "trigger_config cannot be None."
 
     @abstractmethod
-    def inform(self, new_data: list[tuple[int, int, int]]) -> list[int]:
-        """The supervisor informs the trigger about new data.
-        In case the concrete trigger implementation decides to trigger, we return a list of _indices into new_data_.
+    def init_trigger(self, pipeline_id: int, pipeline_config: dict, modyn_config: dict, base_dir: pathlib.Path) -> None:
+        """The supervisor initializes the concrete Trigger with Trigger-type-specific configurations
+        base_dir: the base directory to store Trigger outputs. A location at the supervisor.
+        """
+
+    @abstractmethod
+    def inform(self, new_data: list[tuple[int, int, int]]) -> Generator[int, None, None]:
+        """The supervisor informs the Trigger about new data.
+        In case the concrete Trigger implementation decides to trigger, we return a list of _indices into new_data_.
         This list contains the indices of all data points that cause a trigger.
-        The list might be empty or only contain a single element, which concrete triggers need to respect.
+        The list might be empty or only contain a single element, which concrete Triggers need to respect.
 
              Parameters:
                      new_data (list[tuple[str, int, int]]): List of new data (keys, timestamps, labels). Can be empty.
 
              Returns:
                      triggering_indices (list[int]): List of all indices that trigger training
         """
+
+    @abstractmethod
```

Comment:
Same comment as above - can probably be non abstract methods to avoid empty overrides

Reply:
done